prow/cmd/branchprotector/protect.go

```diff
@@ -211,26 +211,19 @@ func (p *Protector) Protect() {
 		}
 		orgName := parts[0]
 		repoName := parts[1]
-		repoReqs := repoRequirements(orgName, repoName, *p.cfg)
-		protect := len(repoReqs) > 0
-		if err := p.UpdateRepo(orgName, repoName, config.Repo{}, &protect, repoReqs, nil); err != nil {
+		protect := true
+		if err := p.UpdateRepo(orgName, repoName, config.Repo{}, &protect); err != nil {
 			p.errors.add(err)
 		}
 	}
 }
 
 // Update all repos in the org with the specified defaults
-func (p *Protector) UpdateOrg(orgName string, org config.Org, protect *bool, contexts, pushers []string) error {
+func (p *Protector) UpdateOrg(orgName string, org config.Org, protect *bool) error {
```

Comment:
The `protect` no longer needs to be a pointer, and it needs a new name as it no longer serves its prior role of determining whether protection is on or off (just whether we list all children or just defined ones)

Reply:
I did not change that. Did you ?

Reply:
It used to be a pointer because it needed to distinguish between `I don't care (nil)`, `do not protect (false)` and `protect (true)`.

This is no longer what this variable does

Reply:
the getbranchprotection has this information but we still need to traverse the tree to figure out what are the branches we need to apply protection to. The config itself has a pointer for protect, so changing that would be an incompatible change. In anyway I don't think this should be blocking the PR.